ovos_phal_plugin_ipgeo/__init__.py

```diff
@@ -62,9 +65,19 @@ def ip_geolocate(ip=None):
         try:
             # configured backend may throw some errors if its down
             api = GeolocationApi()
+        except Exception as e:
+            LOG.exception("Failed to create Geolocation API")
+            api = GeolocationApi(backend_type=BackendType.OFFLINE)
+        try:
             return api.get_ip_geolocation(ip)
         except Exception as e:
             LOG.exception("Backend Geolocation API error!")
+        try:
             # force offline backend api (direct call)
-            api = GeolocationApi(backend_type=BackendType.OFFLINE)
-            return api.get_ip_geolocation(ip)
+            if api.backend_type != BackendType.OFFLINE:
+                return (GeolocationApi(backend_type=BackendType.OFFLINE)
+                        .get_ip_geolocation(ip))
+        except Exception as e:
+            LOG.error(e)
```

Comment:
suggestion(non-blocking): Consider `LOG.exception()`, which will put the traceback in the logs

Reply:
I changed this to `error` since the exception is re-raised and logged (line 57)